dev/ts/shared/internal-helpers.ts

```diff
@@ -16,34 +16,33 @@ limitations under the License.
 
 import { log } from './shared-helpers';
 
-export function _log(header: string, message: unknown, ...optionalParams: unknown[]): void {
+export function _log(header: string, message: string, ...optionalParams: unknown[]): void {
   // eslint-disable-next-line no-console
-  console.info(header + ' ' + message, ...optionalParams);
+  console.info(`${header} ${message}`, ...optionalParams);
 }
 
 export function logStartup(message?: string, ...optionalParams: unknown[]): void {
   _log('[DEMO STARTUP]', message, ...optionalParams);
 }
 
-export function logErrorAndOpenAlert(error: unknown, alertMsg?: string): void {
+export function logErrorAndOpenAlert(error: string, alertMsg?: string): void {
   logError(error);
   window.alert(alertMsg ?? error);
 }
 
-export function logError(error: unknown): void {
+export function logError(error: string): void {
   console.error(`[DEMO]`, error);
 }
 
-export function logDownload(message?: unknown, ...optionalParams: unknown[]): void {
+export function logDownload(message?: string, ...optionalParams: unknown[]): void {
   _log('[DEMO DOWNLOAD]', message, ...optionalParams);
 }
 
-export function fetchBpmnContent(url: string): Promise<string> {
+export async function fetchBpmnContent(url: string): Promise<string> {
   log(`Fetching BPMN content from url ${url}`);
-  return fetch(url).then(response => {
-    if (!response.ok) {
-      throw Error(`HTTP status ${response.status}`);
-    }
-    return response.text();
-  });
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw Error(`HTTP status ${response.status}`);
+  }
+  return await response.text();
```

Comment:
**thought:** let's discuss this.
We don't want to await here. Here is how it is used 👇🏿 

https://github.com/process-analytics/bpmn-visualization-js/blob/2ac896c84a11c2921a72a3c131818b26ecb73f23/dev/ts/shared/main.ts#L184-L203

Reply:
As the error from `fetch()` and `text()` calling is handling on the function who calls `fetchBpmnContent()`, the new implementation `async/await` is totally appropriate. It follows best practices for using **`async/await`** in asynchronous operations.

#### Examples

*   [https://kentcdodds.com/blog/using-fetch-with-type-script#using-the-typed-fetchpokemon](https://kentcdodds.com/blog/using-fetch-with-type-script#using-the-typed-fetchpokemon)
*   https://www.newline.co/@bespoyasov/how-to-use-fetch-with-typescript--a81ac257

Reply:
This was not a real issue, TypeScript suggested a conversion which has been applied here.`TS80006: This may be converted to an async function.` This has no relation with an eslint rules requesting a change.

Let's keep it and move forward as this is demo code. But we lost a large amount of time on this topic because it was initially expressed as a change related to a eslint newly introduced.